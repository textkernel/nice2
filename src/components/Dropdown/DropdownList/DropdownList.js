import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './DropdownList.scss';

const { block } = bem({
    name: 'DropdownList',
    classnames: styles,
    propsToMods: ['animated', 'fromRight', 'shown']
});

const DropdownList = forwardRef((props, ref) => {
    const { animated, children, fromRight, shown, ...rest } = props;

    if (!shown) {
        return null;
    }

    return (
        <div {...rest} {...block(props)} ref={ref}>
            {children}
        </div>
    );
});

DropdownList.propTypes = {
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]),
    fromRight: PropTypes.bool,
    shown: PropTypes.bool
};

DropdownList.defaultProps = {
    children: null,
    fromRight: false,
    shown: false
};

export default DropdownList;
