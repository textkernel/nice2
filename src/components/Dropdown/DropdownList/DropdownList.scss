.DropdownList {
    background-color: var(--color-background);
    border: {
        color: var(--color-neutral);
        radius: var(--border-radius);
        style: solid;
        width: var(--line-normal);
    }
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.1);
    box-sizing: border-box;
    display: none;
    margin-top: var(--spacing-normal);
    min-width: 100%;
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 2;

    &--animated {
        animation: listIn .1s linear;
    }

    &--shown {
        display: block;
    }

    &--fromRight {
        left: auto;
        right: 0;
    }

    > :first-child {
        border-top: {
            left-radius: var(--border-radius);
            right-radius: var(--border-radius);
        }
    }
    > :last-child {
        border-bottom: {
            left-radius: var(--border-radius);
            right-radius: var(--border-radius);
        }
    }
}
