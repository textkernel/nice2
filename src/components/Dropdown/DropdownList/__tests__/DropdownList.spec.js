import React from 'react';
import toJson from 'enzyme-to-json';
import DropdownList from '../DropdownList';

describe('<DropdownList> that renders a dropdown list', () => {
    it('should render a default list correctly', () => {
        const wrapper = shallow(<DropdownList shown>Some content</DropdownList>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render props correctly', () => {
        const wrapper = shallow(
            <DropdownList animated fromRight shown>
                Some content
            </DropdownList>
        );
        expect(wrapper.hasClass('DropdownList--animated')).toBe(true);
        expect(wrapper.hasClass('DropdownList--fromRight')).toBe(true);
        expect(wrapper.hasClass('DropdownList--shown')).toBe(true);
    });

    it('should render nothing when hidden', () => {
        const wrapper = shallow(<DropdownList>Some content</DropdownList>);
        expect(wrapper.type()).toEqual(null);
    });
});
