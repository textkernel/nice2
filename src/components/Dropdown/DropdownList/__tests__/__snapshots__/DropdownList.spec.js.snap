// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownList> that renders a dropdown list should render a default list correctly 1`] = `
<div
  className="DropdownList DropdownList--shown"
>
  Some content
</div>
`;
